src/ck3/data/artifacts.rs

```diff
@@ -257,3 +258,45 @@ impl DbKind for ArtifactFeatureGroup {
         let mut _vd = Validator::new(block, data);
     }
 }
+
+#[derive(Clone, Debug)]
+pub struct ArtifactBlueprint {}
+
+inventory::submit! {
+    ItemLoader::Normal(GameFlags::Ck3, Item::ArtifactBlueprint, ArtifactBlueprint::add)
+}
+
+impl ArtifactBlueprint {
+    pub fn add(db: &mut Db, key: Token, block: Block) {
+        db.add(Item::ArtifactBlueprint, key, block, Box::new(Self {}));
+    }
+}
+
+impl DbKind for ArtifactBlueprint {
+    fn validate(&self, _key: &Token, block: &Block, data: &Everything) {
+        let mut vd = Validator::new(block, data);
+
+        vd.req_field("in_type");
+        vd.req_field("in_visuals");
+        vd.field_item("in_type", Item::ArtifactType);
+        vd.field_item("in_visuals", Item::ArtifactVisual);
+
+        vd.req_field("out_type");
+        vd.req_field("out_visuals");
+        vd.field_item("out_type", Item::ArtifactType);
+        vd.field_item("out_visuals", Item::ArtifactVisual);
+
+        vd.field_validated_list("disallowed_modifiers", |token, data| {
+            verify_modif_exists(token, data, ModifKinds::all(), Severity::Warning);
```

Comment:
Consider changing this to `ModifKinds::Character`

Reply:
> Consider changing this to `ModifKinds::Character`

Hmm that's a good point, it's possible that artifact bonuses all have to apply to characters.